packages/babel-cli/src/babel/dir.js

```diff
@@ -75,11 +85,20 @@ export default async function({ cliOptions, babelOptions }) {
   async function handleFile(src, base) {
     const written = await write(src, base);
 
-    if (!written && cliOptions.copyFiles) {
+    if (written === NOT_COMPILABLE) {
+      if (!cliOptions.copyFiles) return false;
       const filename = path.relative(base, src);
       const dest = getDest(filename, base);
+      if (cliOptions.incremental) {
+        try {
+          const srcStat = fs.statSync(src);
+          const destStat = fs.statSync(dest);
+          if (srcStat.ctimeMs < destStat.ctimeMs) return false;
+        } catch (e) {}
```

Comment:
We could move this duplicated logic to a `isSrcNewer(src, dest)` helper.

Reply:
Sure.